plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -242,144 +242,204 @@ public function add_fields_data() {
 	 * @return \WP_Error|void True if the field was registered, a WP_Error otherwise.
 	 */
 	public function register_checkout_field( $options ) {
+
+		// Check the options and show warnings if they're not supplied. Return early if an error that would prevent registration is encountered.
+		$result = $this->validate_options( $options );
+		if ( false === $result ) {
+			return;
+		}
+
+		// The above validate_options function ensures these options are valid. Type might not be supplied but then it defaults to text.
+		$id       = $options['id'];
+		$location = $options['location'];
+		$type     = $options['type'] ?? 'text';
+
+		$field_data = array(
+			'label'         => $options['label'],
+			'hidden'        => false,
+			'type'          => $type,
+			'optionalLabel' => empty( $options['optionalLabel'] ) ? sprintf(
+			/* translators: %s Field label. */
+				__( '%s (optional)', 'woocommerce' ),
+				$options['label']
+			) : $options['optionalLabel'],
+			'required'      => empty( $options['required'] ) ? false : $options['required'],
+		);
+
+		$field_data['attributes'] = $this->register_field_attributes( $id, $options['attributes'] ?? [] );
+
+		if ( 'checkbox' === $type ) {
+			$result = $this->process_checkbox_field( $options, $field_data );
+
+			// $result will be false if an error that will prevent the field being registered is encountered.
+			if ( false === $result ) {
+				return;
+			}
+			$field_data = $result;
+		}
+
+		if ( 'select' === $type ) {
+			$result = $this->process_select_field( $options, $field_data );
+
+			// $result will be false if an error that will prevent the field being registered is encountered.
+			if ( false === $result ) {
+				return;
+			}
+			$field_data = $result;
+		}
+
+		// Insert new field into the correct location array.
+		$this->additional_fields[ $id ]        = $field_data;
+		$this->fields_locations[ $location ][] = $id;
+	}
+
+	/**
+	 * Validates the "base" options (id, label, location) and shows warnings if they're not supplied.
+	 *
+	 * @param array $options The options supplied during field registration.
+	 * @return bool false if an error was encountered, true otherwise.
+	 */
+	private function validate_options( $options ) {
 		if ( empty( $options['id'] ) ) {
 			_doing_it_wrong( 'woocommerce_blocks_register_checkout_field', 'A checkout field cannot be registered without an id.', '8.6.0' );
```

Comment:
Shouldn't we make all `_doing_it_wrong` messages translatable? For example, in the `CheckoutFields` class, we're also making the error messages translatable:
```php
if ( ! $this->is_field( $key ) ) {
	return new \WP_Error(
		'woocommerce_blocks_checkout_field_invalid',
		\sprintf(
			// translators: % is field key.
			__( 'The field %s is invalid.', 'woocommerce' ),
			$key
		)
	);
}
```

Reply:
Developer errors shouldn't be translatable, but let me confirm again.

Reply:
Yep, WordPress doesn't translate any of the `doing_it_wrong` messages
https://github.com/search?q=repo:WordPress/WordPress%20doing_it_wrong&type=code

Reply:
The difference between this message and the one Niels linked to is the one Niels linked is user-facing - the `_doing_it_wrong` messages are actually only admin/dev-facing.

Reply:
> Yep, WordPress doesn't translate any of the doing_it_wrong messages
https://github.com/search?q=repo:WordPress/WordPress%20doing_it_wrong&type=code

I see various `_doing_it_wrong` calls with translated messages on the link you shared, e.g. 
- https://github.com/WordPress/WordPress/blob/3a36e92c88a35358a130024c4b7667f26788b726/wp-includes/nav-menu.php#L97
- https://github.com/WordPress/WordPress/blob/3a36e92c88a35358a130024c4b7667f26788b726/wp-includes/customize/class-wp-customize-partial.php#L226
- https://github.com/WordPress/WordPress/blob/3a36e92c88a35358a130024c4b7667f26788b726/wp-includes/query.php#L165

Reply:
The i18n said it's okay not to translate developer errors, but if they surface to customers, they should be translated, as long as the error code is not translated. Which is a bit confusing.

https://a8c.slack.com/archives/C02AED43D/p1705666614969519

> In general, errors are not translated, unless we have both the error key/code for handing the error and the error description which the user could see.

> [regarding the PR] Since by default it triggers a PHP notice and it also adds some translated debugging information, the message itself shouldn’t be translated.

Reply:
Those errors aren't user facing so I think it's fine.

Reply:
Thanks for double-checking this, @senadir. In this case, I'll mark my comment as resolved, @opr.